src/components/tl/mlx5/mcast/tl_mlx5_mcast_allgather.c

```diff
@@ -433,6 +429,195 @@ ucc_tl_mlx5_mcast_prepare_zero_copy_allgather(ucc_tl_mlx5_mcast_coll_comm_t *com
     return UCC_OK;
 }
 
+static inline ucc_status_t
+ucc_tl_mlx5_mcast_check_zcopy_collective(ucc_tl_mlx5_mcast_coll_comm_t *comm,
+                                         ucc_tl_mlx5_mcast_coll_req_t  *req)
+{
+    ucc_tl_mlx5_mcast_pipelined_ag_schedule_t *sched = req->ag_schedule;
+    ucc_status_t                               status;
+
+    if (!sched[req->step].prepost_buf_op_done || !sched[req->step].multicast_op_done) {
+        // it is not yet the time to start the reliability protocol
+        return UCC_INPROGRESS;
+    }
+    if (comm->one_sided.pending_reads) {
+        return ucc_tl_mlx5_mcast_progress_one_sided_communication(comm, req);
+    }
+    if (req->allgather_rkeys_req) {
+        status = comm->service_coll.coll_test(req->allgather_rkeys_req);
+        if (status != UCC_OK) {
+            return status;
+        }
+        req->allgather_rkeys_req = NULL;
```

Comment:
These must be different rkeys than the ones sued in ucc_tl_mlx5_mcast_progress_one_sided_communication, right? What's the difference between them?

Reply:
They both are related to rkey/addr of the send buffer of the remote processes.